client/my-sites/posts/main.jsx

```diff
@@ -34,21 +34,11 @@ import {
 import { getEditorNewPostPath } from 'state/ui/editor/selectors';
 import { warningNotice } from 'state/notices/actions';
 import {
-	getSiteAdminUrl,
 	getSiteSlug,
 	isJetpackSite,
-	siteHasMinimumJetpackVersion
 } from 'state/sites/selectors';
 
 const PostsMain = React.createClass( {
-	componentWillMount() {
-		this.setWarning( this.props );
-	},
-
-	componentWillReceiveProps( nextProps ) {
-		this.setWarning( nextProps );
```

Comment:
This is likely the source of the problem and a fix could be applied here. Perhaps just checking `this.props` and `nextProps` equality for the props relevant to showing the warning: `siteId`, `isJetpack`, `hasMinimumJetpackVersion`.